src/worker.cc

```diff
@@ -1,297 +1,140 @@
-
-    {120, 400}, {121, 400}, {122, 400}, {123, 400}, {124, 400}, {125, 400}, {126, 400}, {127, 400},
-
-    {130, 400}, {131, 400}, {132, 400}, {133, 400}, {136, 400}, {137, 400},
-
-    {140, 400}, {141, 501}, {142, 501}, {143, 400},
-
-    {150, 400}, {151, 400}, {152, 400}, {153, 400}, {154, 400}, {155, 400}, {156, 400}, {157, 400},
-    {158, 400}, {159, 400},
-
-    {160, 400}, {161, 400}, {162, 400}, {163, 400}, {164, 400}, {165, 400}, {166, 400}, {167, 400},
-
-    {170, 400}, {171, 400}, {172, 400},
-
-    {199, 400},
-
-    {200, 500}, {201, 500}, {202, 500}, {203, 503},
-
-    {210, 400}, {211, 400}, {212, 400}, {213, 400},
 
-    {220, 400},
-
-    {230, 400}, {231, 400}, {232, 400},
-
-    {299, 400},
-
-    {304, 404}, {305, 501},
-
-    {310, 400}, {311, 400}, {312, 400}, {313, 400}, {314, 400},
-
-    {399, 400},
-
-    {400, 400}, {401, 500}, {402, 503},
-
-    {420, 400}, {421, 400}, {422, 400}, {423, 400}, {424, 400},
-
-    {430, 400},
-
-    {440, 400}, {441, 400}, {442, 400}, {443, 400}, {444, 400}, {445, 400},
-
-    {499, 400},
-
-    {500, 500}, {501, 500}, {502, 400},
-
-    {599, 400},
+// clang-format off
+constexpr const char* HTTP_400 = "Bad Request";
+constexpr const char* HTTP_404 = "Not Found";
+constexpr const char* HTTP_405 = "Method Not Allowed";
+constexpr const char* HTTP_500 = "Internal Server Error";
+constexpr const char* HTTP_501 = "Not Implemented";
+constexpr const char* HTTP_503 = "Service Unavailable";
+constexpr const char* OSRM_INVALID_URL = R"({"code":"InvalidUrl","message":"URL string is invalid."})";
+constexpr const char* OSRM_INVALID_SERVICE = R"({"code":"InvalidService","message":"Service name is invalid."})";
+constexpr const char* OSRM_INVALID_OPTIONS = R"({"code":"InvalidOptions","message":"Options are invalid."})";
+constexpr const char* OSRM_INVALID_VALUE = R"({"code":"InvalidValue","message":"The successfully parsed query parameters are invalid."})";
+constexpr const char* OSRM_NO_ROUTE = R"({"code":"NoRoute","message":"Impossible route between points"})";
+constexpr const char* OSRM_NO_SEGMENT = R"({"code":"NoSegment","message":"One of the supplied input coordinates could not snap to street segment."})";
+constexpr const char* OSRM_SHUTDOWN = R"({"code":"ServiceUnavailable","message":"The service is shutting down."})";
+constexpr const char* OSRM_SERVER_ERROR = R"({"code":"InvalidUrl","message":"Failed to serialize route."})";
+constexpr const char* OSRM_DISTANCE_EXCEEDED = R"({"code":"DistanceExceeded","message":"Path distance exceeds the max distance limit."})";
+constexpr const char* OSRM_PERIMETER_EXCEEDED = R"({"code":"PerimeterExceeded","message":"Perimeter of avoid polygons exceeds the max limit."})";
+constexpr const char* OSRM_BREAKAGE_EXCEEDED = R"({"code":"BreakageDistanceExceeded","message":"All coordinates are too far away from each other"})";
+
+using ve = valhalla_exception_t;
+const std::unordered_map<unsigned, valhalla::valhalla_exception_t> error_codes{
```

Comment:
so we had a bunch of parallel maps and it was a real mental burdon to keep them synchronized not to mention look up which ones went with eachother. so instead of maintaining that, ive revamped the `valhalla_exception_t` class to store all the different types of info directly and declare all of those predefined (ie non custom) ones here. i turned off clang format because it just brutally destroys the formatting and makes it again very h ard to read. 1 line, 1 error and sanity remains.